src/wallet/rpcwallet.cpp

```diff
@@ -3672,9 +3673,9 @@ class DescribeWalletAddressVisitor : public boost::static_visitor<UniValue>
             // Always report the pubkey at the top level, so that `getnewaddress()['pubkey']` always works.
             if (subobj.exists("pubkey")) obj.pushKV("pubkey", subobj["pubkey"]);
             obj.pushKV("embedded", std::move(subobj));
-        } else if (which_type == TxoutType::MULTISIG) {
-            // Also report some information on multisig scripts (which do not have a corresponding address).
-            // TODO: abstract out the common functionality between this logic and ExtractDestinations.
+        } else if (find(enabled_methods.begin(), enabled_methods.end(), "reqSigs") != enabled_methods.end() &&
+                   which_type == TxoutType::MULTISIG) {
+            // TODO: from v0.22 ("addresses" and "reqSigs" deprecated) this entire else if statement should be removed
```

Comment:
Note that the next major release after 0.21 is going to be v22 (not v0.22); see https://github.com/bitcoin/bitcoin/pull/20223.